packages/assets-controllers/src/MultichainBalancesController/Poller.ts

```diff
@@ -0,0 +1,30 @@
+export class Poller {
+  #interval: number;
+
+  #callback: () => Promise<void>;
+
+  #handle: NodeJS.Timeout | undefined = undefined;
+
+  constructor(callback: () => Promise<void>, interval: number) {
+    this.#interval = interval;
+    this.#callback = callback;
+  }
+
+  start() {
+    if (this.#handle) {
+      return;
+    }
+
+    this.#handle = setInterval(() => {
+      this.#callback().catch(console.error);
```

Comment:
I was thinking about the error handling, and I think we could use this pattern:

```ts
class PollerError extends Error {
  constructor(cause: Error) {
    super('Polling error', { cause });
    this.name = 'PollerError';
  }
}
```

and then use it like:

```suggestion
      this.#callback().catch((error) => {
        console.error(new PollerError(error));
      });
```

Which should show something like this:

```
Uncaught PollerError: Polling error:
  at <anonymous>:1:7
Caused by: Error: ...
  at <anonymous>:1:13
```

WDYT?

Reply:
Changes in [fe5d9ec](https://github.com/MetaMask/core/pull/4965/commits/fe5d9ec31b94bbec89d72e1e86593a24c2b7e43c) and [b355d11](https://github.com/MetaMask/core/pull/4965/commits/b355d119270e80f01195d9fc67e36651c9711033)